request_handler/request_handler.go

```diff
@@ -90,23 +90,24 @@ func auditTrail(e *spade.Event) string {
 	return fmt.Sprintf("[%d] %s", e.ReceivedAt.Unix(), e.Uuid)
 }
 
-func getIpFromHeader(headerKey string, header http.Header) net.IP {
-	clientIp := header.Get(headerKey)
-	comma := strings.Index(clientIp, ",")
-	if comma > -1 {
-		clientIp = clientIp[:comma]
+func parseLastForwarder(header string) net.IP {
+	var clientIp string
+	comma := strings.LastIndex(header, ",")
+	if comma > -1 && comma < len(header)+1 {
+		clientIp = header[comma+1:]
+	} else {
+		clientIp = header
 	}
 
-	return net.ParseIP(clientIp)
+	return net.ParseIP(strings.TrimSpace(clientIp))
 }
 
 func (s *SpadeHandler) HandleSpadeRequests(r *http.Request, context *requestContext) int {
 	statTimer := newTimerInstance()
 
-	clientIp := getIpFromHeader(context.IpHeader, r.Header)
-	if clientIp == nil {
-		return http.StatusBadRequest
-	}
+	xForwardedFor := r.Header.Get(context.IpHeader)
+	clientIp := parseLastForwarder(xForwardedFor)
```

Comment:
So is it guaranteed that some IP will show up in the line? (you removed the == nil test)